ground/src/main/java/com/google/android/ground/coroutines/CoroutineDispatchersModule.kt

```diff
@@ -0,0 +1,48 @@
+/*
```

Comment:
Do we really need this? Why wouldn't we just write:

```kotlin
someCoroutineScope.launch(Dispatchers.IO) { 
 //...
}

// and
suspend fun foo() {
  // ...
  return withContext(Dispatchers.IO) {
      // other stuff
  }
}
```

Having this explicitly written in source seems clearer to me than relying on injection. Injection makes the code harder to follow and now we have more dagger magic to maintain in the codebase. While this gives us a little bit more modularity and flexibility, we're tied to the Android platform anyway and I think it might be a premature architectural optimization. I'd prefer to go with less code in this case and use the platform directly.

Reply:
Yes! Please check out https://developer.android.com/kotlin/coroutines/coroutines-best-practices for motivations. tldr, we need to be able to replace these for tests.

Reply:
Ah! Ok! sgtm then. 

> Note: While you might have seen hardcoded dispatchers in code snippets across this site and our codelabs, this is only to keep the sample code concise and simple. In your application, you should inject dispatchers.

Gotta love it.